### ISSUE TITLE
Lightning Gateway SCID should be random or contain more identification info

### ISSUE THREAD

**Issue by Kodylow** [fedimint/fedimint#4085]
We've thought about this previously but it came up today again with the API integrations where you might have multiple gateways using a single lightning node or lightning service provider. Our model of using scids 00, 01, 02 only works for 1-to-1 node to gateway. The moment we want to use sub accounts on the node (in the coinos case custodial accounts but this would also affect using LND macaroon based subaccounts I think) you need something else to map the scid to the correct gateway.

Is the scid the only place we can hide info for these purposes? Could we also maybe use the recipient pubkey or is that already used for contract identification by the gateway?

**Comment by m1sterc001guy** [comment]
Recipient pub key is currently just random:

```
        // Temporary lightning node pubkey
        let (node_secret_key, node_public_key) = self.secp.generate_keypair(&mut rng);
```

Perhaps we should just encode the `federation_id` and use it as the recipient pubkey? We already do something like this for internal transfers within the mint (we put it in the route hints). I suppose this has the downside that the invoice will reveal the Fedimint that is being paid (same way that lightning invoices reveal recipient's node pubkey), but in Fedimint's case it obviously doesn't reveal the individual user. 

I think regardless we should probably be using random scids anyway.

**Comment by Kodylow** [comment]
Agreed, using the pubkey to hold additional useful information for the gateway would be great, like we could use the pubkey for an actual user.

For random scid does that have rules around not colliding with real channels?

**Comment by elsirion** [comment]
> Could we also maybe use the recipient pubkey

We cannot choose the pub key freely as we have to sign the invoice using the corresponding secret key.
